Packages/org.plunderludics.UnityHawk/Runtime/Emulator/Emulator.cs

```diff
@@ -424,18 +290,34 @@ public void OnEnable()
             _emuhawk.StartInfo.UseShellExecute = false;
         }
 
-        if (firmwareDirFullPath != null) {
-            args.Add($"--firmware={firmwareDirFullPath}");
+        // add rom path
+        args.Add(Paths.GetAssetPath(romFile));
+
+        // add config path
+        if (configFile) {
+            args.Add($"--config={Paths.GetAssetPath(configFile)}");
+        }
+
+        // add save state path
+        if (saveStateFile) {
+            args.Add($"--load-state={Paths.GetAssetPath(saveStateFile)}");
         }
 
-        args.Add($"--savestates={savestatesOutputDirFullPath}");
+        // Save savestates with extension .savestate instead of .State, this is because Unity treats .State as some other kind of asset
+        args.Add($"--savestate-extension={_savestateExtension}");
+
+        args.Add($"--savestates={Paths.SavestatesOutputPath}");
+
+        // add firmware
+        args.Add($"--firmware={Paths.FirmwarePath}");
```

Comment:
Have you tested if this breaks when the directory doesn't exist?

Reply:
if Paths.SavestatesOutputPath doesnt exists? no

Reply:
Oh I meant firmware